python/langsmith/client.py

```diff
@@ -145,16 +123,44 @@ def _serialize_json(obj: Any) -> str:
         return str(obj)
 
 
-def close_session(session: requests.Session) -> None:
+def _get_default_retry_config() -> dict:
+    return {
+        "stop": stop_after_attempt(3),
+        "wait": wait_exponential_jitter(initial=0.5),
+        # Retry only on certain status codes ([502, 503, 504, 408, 425, 429])
+        "retry": retry_if_exception_type(ls_utils.LangSmithRetryableError),
+        "before_sleep": before_sleep_log(logger, logging.DEBUG),
+    }
+
+
+def close_client(client: httpx.Client) -> None:
     """Close the session.
 
     Parameters
     ----------
-    session : Session
-        The session to close.
+    client : client
+        The httpx client to close.
     """
-    logger.debug("Closing Client.session")
-    session.close()
+    if client.is_closed:
+        return
+    logger.debug("Closing Client._client")
+    client.close()
+
+
+def close_async_client(client: httpx.AsyncClient) -> None:
+    logger.debug("Closing Client._aclient")
+    if client.is_closed:
+        return
+    try:
+        with concurrent.futures.ThreadPoolExecutor(1) as executor:
+            executor.submit(lambda: asyncio.run(client.aclose())).result(timeout=10)
```

Comment:
This may sometime lock if called explicitly in our tests if garbage collection isn't triggered explicitly beforehand (not always but sometimes). I'm not sure why at the moment.